jar-infer/jar-infer-lib/src/main/java/com/uber/nullaway/jarinfer/BytecodeAnnotator.java

```diff
@@ -63,21 +89,63 @@ private static void LOG(boolean cond, String tag, String msg) {
   private static final String DIGEST_ENTRY_PATTERN =
       "Name: [A-Za-z0-9/\\$\\n\\s\\-\\.]+[A-Za-z0-9]\\nSHA-256-Digest: " + BASE64_PATTERN;
 
-  private static void addAnnotationIfNotPresent(
-      List<AnnotationNode> annotationList, String annotation) {
-    for (AnnotationNode node : annotationList) {
-      if (node.desc.equals(annotation)) {
-        return;
+  private static boolean annotationsShouldBeVisible(String nullableDesc) {
+    if (nullableDesc.equals(javaxNullableDesc)) {
+      return true;
+    } else if (nullableDesc.equals(androidNullableDesc)) {
+      return false;
+    } else {
+      throw new Error("Unknown nullness annotation visibility");
+    }
+  }
+
+  private static boolean listHasNullnaessAnnotations(List<AnnotationNode> annotationList) {
```

Comment:
typo: Nullness

Reply:
Wow. Lol. No idea where that 'a' came from...